File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/EnTransRepository.java
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.dao.EnTransDao;
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.db.EnTransDatabase;
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.EnTrans;

public class EnTransRepository {
    private EnTransDatabase enTransDatabase;
    private EnTransDao enTransDao;

    public EnTransRepository(Application application) {
        enTransDatabase = EnTransDatabase.getInstance(application);
        enTransDao = enTransDatabase.enTransDao();
    }

    public LiveData<EnTrans> getEnTrans(int id) {
        return enTransDao.getEnTrans(id);
    }

    public LiveData<List<EnTrans>> getAllEnTrans() {
        return enTransDao.fetchAllEnTrans();
    }
}

File: app/src/main/java/id/ac/ui/cs/mobileprogramming/nabilafv/rememberyourlord/repository/EnTransRepository.kt
package id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.repository

import androidx.lifecycle.LiveData
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.dao.EnTransDao
import id.ac.ui.cs.mobileprogramming.nabilafv.rememberyourlord.model.EnTrans
import javax.inject.Inject

class EnTransRepository @Inject constructor(private val enTransDao: EnTransDao) :
    EnTransRepositoryInterface {
    override fun getEnTrans(id: Int): EnTrans {
        return enTransDao.getEnTrans(id)
    }

    override fun getAllEnTrans(): List<EnTrans> {
        return enTransDao.fetchAllEnTrans()
    }
}